File: Loenn/triggers/multilayer_music_fade_trigger.lua
local enums = require("consts.celeste_enums")

local multilayer_music_fade_trigger = {}

multilayer_music_fade_trigger.name = "ShroomHelper/MultilayerMusicFadeTrigger"

multilayer_music_fade_trigger.fieldInformation = {
    P1Direction = {
        options = enums.trigger_position_modes,
        editable = false
    },
    P2Direction = {
        options = enums.trigger_position_modes,
        editable = false
    },
    P3Direction = {
        options = enums.trigger_position_modes,
        editable = false
    }
}

multilayer_music_fade_trigger.placements = {
    name = "multilayer_music_fade_trigger",
    data = {
        trackEvent = "",
        P1 = "",
        P2 = "",
        P3 = "",
        P1From = 1.0,
        P2From = 1.0,
        P3From = 1.0,
        P1To = 1.0,
        P2To = 1.0,
        P3To = 1.0,
        P1Direction = "LeftToRight",
        P2Direction = "LeftToRight",
        P3Direction = "LeftToRight",
        persistent = false,
        destroyOnLeave = false
    }
}

return multilayer_music_fade_trigger

File: Ahorn/triggers/multilayerMusicFadeTrigger.jl
module ShroomHelperMultilayerMusicFadeTrigger

using ..Ahorn, Maple

@mapdef Trigger "ShroomHelper/MultilayerMusicFadeTrigger" MultilayerMusicFadeTrigger(
    x::Integer, 
    y::Integer, 
    width::Integer=Maple.defaultTriggerWidth, 
    height::Integer=Maple.defaultTriggerHeight,
    trackEvent::String="",
    P1::String="",
    P2::String="",
    P3::String="",
    P1From::Number=1.0, 
    P2From::Number=1.0, 
    P3From::Number=1.0, 
    P1To::Number=1.0, 
    P2To::Number=1.0, 
    P3To::Number=1.0, 
    P1Direction::String="LeftToRight",
    P2Direction::String="LeftToRight",
    P3Direction::String="LeftToRight",
    persistent::Bool=false,
    destroyOnLeave::Bool=false,
)

const placements = Ahorn.PlacementDict(
    "Multilayer Music Fade Trigger (Shroom Helper)" => Ahorn.EntityPlacement(
        MultilayerMusicFadeTrigger,
        "rectangle"
    )
)

Ahorn.editingOptions(entity::MultilayerMusicFadeTrigger) = Dict{String, Any}(
    "P1Direction" => Maple.trigger_position_modes,
    "P2Direction" => Maple.trigger_position_modes,
    "P3Direction" => Maple.trigger_position_modes
)

end
